src/addDefaultParameters.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  Codacyrc,
+  Parameter,
+  ParameterSpec,
+  Pattern,
+  PatternSpec,
+  Specification
+} from "."
+
+export function addDefaultParameters(
```

Comment:
I would change this function to `withDefaultParameters` that returns a new object with the added parameters in it here

Reply:
Yes when changing the Pattern to be immutable, I will be changing this. But why the "with" keyword? is this a standard that we follow?

Reply:
`withThis` is a `not-so-common` way to name immutable builder pattern methods.
I suggested this because `addDefaultParameters` means you're adding something, but if you return a changed object you're not adding anything to the original object. At least this is the rationale :D

Reply:
I was actually think that "add" returns the thing this the added fields if the function has it as return type. Like in scala you have the same names for mutable or immutable collections, it depends on the return type. But I can use that name if you guys prefer